### ISSUE TITLE
Makefile install errors with parallel jobs

### ISSUE THREAD

**Issue by ivan-pi** [sfilippone/psblas3#23]
I have been facing persistent problems when trying to build and install psblas using multiple jobs. The build phase (`make -j8`) seems to complete successfully.

The errors always originate in commands related to `ranlib`, e.g.

```txt
==> Installing psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3
==> No binary for psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3 found: installing from source
==> Using cached archive: /path/to/cache/_source-cache/archive/dc/dca593010f258af669eb40d92c4934eaaa016ef17280d4ab1208afc3d5aa13f0.tar.gz
==> No patches needed for psblas
==> psblas: Executing phase: 'autoreconf'
==> psblas: Executing phase: 'configure'
==> psblas: Executing phase: 'build'
==> psblas: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'install'

4 errors found in build log:
     2875    ranlib ../../libpsb_base.a
     2876    ranlib ../libpsb_base.a
     2877    ranlib ../../libpsb_base.a
     2878    ranlib ../../libpsb_base.a
     2879    ranlib ../libpsb_base.a
     2880    ranlib ../libpsb_base.a
  >> 2881    ranlib: ../libpsb_base.a: error reading psi_eswapdata_a.o: file truncated
  >> 2882    ranlib: ../libpsb_base.a: error reading psb_cgetmatinfo.o: file truncated
     2883    make[2]: *** [Makefile:45: lib] Error 1
     2884    make[2]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/tools'
     2885    make[1]: *** [Makefile:26: tl] Error 2
     2886    make[1]: *** Waiting for unfinished jobs....
     2887    make[2]: *** [Makefile:29: lib] Error 1
     2888    make[2]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/psblas'
     2889    make[1]: *** [Makefile:24: pb] Error 2
  >> 2890    ranlib: ../libpsb_base.a: error reading psb_z_linmap_mod.o: file truncated
     2891    make[2]: *** [Makefile:21: lib] Error 1
     2892    make[2]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/internals
             '
     2893    make[1]: *** [Makefile:22: in] Error 2
  >> 2894    ranlib: ../../libpsb_base.a: error reading psb_z_linmap_mod.o: file truncated
     2895    make[3]: *** [Makefile:36: lib] Error 1
     2896    make[3]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/comm/inte
             rnals'
     2897    make[2]: *** [Makefile:35: interns] Error 2
     2898    make[2]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/comm'
     2899    make[1]: *** [Makefile:20: cm] Error 2
     2900    make[3]: Leaving directory '/path/to/spack-stage-psblas-3.8.0-1-vuk3jokksrg4k34bby2h23wkbnwg3xr3/spack-src/base/serial/im
             pl'

```

If I use a single job `-j 1` it has worked reliably so far.

**Comment by sfilippone** [comment]
Hmm.
It was supposed to have been fixed, apparently not. Does it fail also with
current development checkout?
Thanks

**Comment by sfilippone** [comment]
Ok, I see, the fixes were put in after the release. Could you please verify that the current development branch works? 
Thanks

**Comment by ivan-pi** [comment]
I repeated the build with the development branch, and it was successful. 🎉

```txt
==> Installing psblas-development-slv73tzxaty3pokyqq44b7at2zx5pjm5
==> No binary for psblas-development-slv73tzxaty3pokyqq44b7at2zx5pjm5 found: installing from source
==> No patches needed for psblas
==> psblas: Executing phase: 'autoreconf'
==> psblas: Executing phase: 'configure'
==> psblas: Executing phase: 'build'
==> psblas: Executing phase: 'install'
==> psblas: Successfully installed psblas-development-slv73tzxaty3pokyqq44b7at2zx5pjm5
  Fetch: 4.28s.  Build: 8m 21.66s.  Total: 8m 25.94s.
[+] /path/to/psblas/development-gcc-11.2.0-slv73tz
```

This also means I have now a (semi-)working Spack package for PSBLAS (I need to add the `-j1` flag for older versions; potentially also compiler version constraints). If you agree/permit I can contribute it to the centralized [Spack](https://github.com/spack/spack) repository. Installing PSBLAS is as simple as `spack install psblas`.

The package currently looks as follows:

```
$ spack info psblas
AutotoolsPackage:   psblas

Description:
    The PSBLAS library, developed with the aim to facilitate the
    parallelization of computationally intensive scientific applications, is
    designed to address parallel implementation of iterative solvers for
    sparse linear systems through the distributed memory paradigm. It
    includes routines for multiplying sparse matrices by dense matrices,
    solving block diagonal systems with triangular diagonal entries,
    preprocessing sparse matrices, and contains additional routines for
    dense matrix operations. The current implementation of PSBLAS addresses
    a distributed memory execution model operating with message passing. The
    PSBLAS library version 3 is implemented in the Fortran 2003 programming
    language, with reuse and/or adaptation of existing Fortran 77 and
    Fortran 95 software, plus a handful of C routines.

Homepage: https://psctoolkit.github.io/products/psblas/

Externally Detectable: 
    False

Tags: 
    linear-equation  matrices  sparse-blas  spmv  vectors

Preferred version:  
    3.8.0-1        https://github.com/sfilippone/psblas3/archive/refs/tags/v3.8.0-1.tar.gz

Safe versions:  
    3.8.0-1        https://github.com/sfilippone/psblas3/archive/refs/tags/v3.8.0-1.tar.gz
    3.8.0          https://github.com/sfilippone/psblas3/archive/refs/tags/v3.8.0.tar.gz
    3.7.1          https://github.com/sfilippone/psblas3/archive/refs/tags/v3.7.1.tar.gz
    3.7.0          https://github.com/sfilippone/psblas3/archive/refs/tags/v3.7.0.tar.gz
    3.6.1          https://github.com/sfilippone/psblas3/archive/refs/tags/v3.6.1.tar.gz
    3.6.0          https://github.com/sfilippone/psblas3/archive/refs/tags/v3.6.0.tar.gz
    development    [git] https://github.com/sfilippone/psblas3.git on branch development

Deprecated versions:  
    None

Variants:
    Name [Default]    When    Allowed values    Description
    ==============    ====    ==============    =============================================================

    amd [off]         --      on, off           Build with AMD package of Davis, Duff and Amestoy
    int64 [on]        --      on, off           Use int64 (8-byte) for global indices of distributed matrices
    metis [on]        --      on, off           Build with METIS support for graph partitioning
    openmp [off]      --      on, off           Enable OpenMP support
    serial [off]      --      on, off           Build serial version with fake MPI stubs

Installation Phases:
    autoreconf    configure    build    install

Build Dependencies:
    autoconf  blas  gnuconfig  lapack  metis  mpi  suite-sparse

Link Dependencies:
    blas  lapack  metis  mpi  suite-sparse

Run Dependencies:
    None

Virtual Packages: 
    None
```

**Comment by sfilippone** [comment]
Yeah, by all means. 
However, I think I will publish an updated release (like 3.8.0-2), so you should use that one. 
It should be done by today.

**Comment by ivan-pi** [comment]
Was it always the case in the past before v3.8, that `psblas` had to be built serially?

I can force this constraint in Spack with something like this (the `'@<from>:<until>'` is the version specifier syntax):
```python
    def build_args(spec,prefix):
        args = []
        if spec.satisfies('@:3.8.0'):
            args.appends('-j1')
        return args 

    def install_args(spec,prefix):
        args = []
        if spec.satisfies('@:3.8.0-1'):
            args.appends('-j1')
        return args
```

**Comment by sfilippone** [comment]
No, it wasn't always the case, I used to be able to build in parallel.
I suspect there are some changes in the environment and/or MAKE that are
causing this issue; fortunately, I could find a workaround modifying the
Makefiles.

S.

**Comment by sfilippone** [comment]
Published 3.8.0-2.
